doctest/test_docs.py

```diff
@@ -0,0 +1,179 @@
+        output = "\n".join(output)
+
+        click.echo(output)
+
+
+def pretty_print(s):
+    try:
+        d = json.loads(s)
+        print(json.dumps(d, indent=2))
+    except json.decoder.JSONDecodeError:
+        print(s)
+
+
+sql_cmd = DocTestConnection(query_language="sql")
+ppl_cmd = DocTestConnection(query_language="ppl")
+test_data_client = Elasticsearch([ENDPOINT], verify_certs=True)
+
+
+def sql_cli_transform(s):
+    return u'sql_cmd.process({0})'.format(repr(s.strip().rstrip(';')))
+
+
+def ppl_cli_transform(s):
+    return u'ppl_cmd.process({0})'.format(repr(s.strip().rstrip(';')))
+
+
+def bash_transform(s):
+    # TODO: add ppl support, be default cli uses sql
+    if s.startswith("odfesql"):
+        s = re.search(r"odfesql\s+-q\s+\"(.*?)\"", s).group(1)
+        return u'cmd.process({0})'.format(repr(s.strip().rstrip(';')))
+    return (r'pretty_print(sh("""%s""").stdout.decode("utf-8"))' % s) + '\n'
+
+
+sql_cli_parser = zc.customdoctests.DocTestParser(
+    ps1='od>', comment_prefix='#', transform=sql_cli_transform)
+
+ppl_cli_parser = zc.customdoctests.DocTestParser(
+    ps1='od>', comment_prefix='#', transform=ppl_cli_transform)
+
+bash_parser = zc.customdoctests.DocTestParser(
+    ps1=r'sh\$', comment_prefix='#', transform=bash_transform)
+
+
+def set_up_accounts(test):
+    set_up(test)
+    load_file("accounts.json", index_name=ACCOUNTS)
+
+
+def load_file(filename, index_name):
+    filepath = "./test_data/" + filename
+
+    # generate iterable data
+    def load_json():
+        with open(filepath, "r") as f:
+            for line in f:
+                yield json.loads(line)
+
+    # Need to enable refresh, because the load won't be visible to search immediately
+    # https://stackoverflow.com/questions/57840161/elasticsearch-python-bulk-helper-api-with-refresh
+    helpers.bulk(test_data_client, load_json(), stats_only=True, index=index_name, refresh='wait_for')
+
+
+def set_up(test):
+    test.globs['sql_cmd'] = sql_cmd
+    test.globs['ppl_cmd'] = ppl_cmd
+
+
+def tear_down(test):
+    # drop leftover tables after each test
+    # TODO: delete all will potentially also delete AES FGAC metadata index
+    test_data_client.indices.delete(index=[ACCOUNTS, EMPLOYEES], ignore_unavailable=True)
+
+
+docsuite = partial(doctest.DocFileSuite,
+                   tearDown=tear_down,
+                   parser=sql_cli_parser,
+                   optionflags=doctest.NORMALIZE_WHITESPACE | doctest.ELLIPSIS,
+                   encoding='utf-8')
+
+
+doctest_file = partial(os.path.join, 'docs')
+
+
+def doctest_files(*items):
+    return (doctest_file(item) for item in items)
+
+
+class DocTests(unittest.TestSuite):
+    def run(self, result, debug=False):
+        super().run(result, debug)
+
+
+def load_tests(loader, suite, ignore):
+    tests = []
+    # docs with bash-based examples
+    for fn in doctest_files('ppl/curl.rst'): # TODO: Add 'sql/explain.rst' after codebase migration
```

Comment:
Does it require the develper change this file every time when they add different rst file?

Reply:
Theoretically, yes. That's because we need different setup, parser creation for docs containing different type of examples. `sql`, `ppl` or `curl`

Mixing sql and ppl code examples together in one doc doesn't work with doctest now, because we setup different client with different query language with CLI.

Reply:
Added `category.json` to help user setup doctest without touching the doctest core code